src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -0,0 +1,513 @@
+        {
+            // check if same topics/partitions have already been requested by someone else,
+            // which is an indication that something is wrong, but it might be alright when assignments change.
+            foreach (var (actorRef, request) in _requests.ToTuples())
+            {
+                if (!actorRef.Equals(fromStage) && request.Topics.Any(topics.Contains))
+                {
+                    _log.Warning($"{updateType} from topic/partition {string.Join(", ", topics)} " +
+                                 $"already requested by other stage {string.Join(", ", request.Topics)}");
+                    actorRef.Tell(new KafkaConsumerActorMetadata.Internal.Messages<K, V>(request.RequestId, ImmutableList<ConsumeResult<K, V>>.Empty));
+                    _requests = _requests.Remove(actorRef);
+                }
+            }
+        }
+
+        private void ReceivePoll(Internal.Poll<K, V> poll)
+        {
+            if (poll.Target == this)
+            {
+                var refreshOffsets = _commitRefreshing.RefreshOffsets;
+                if (refreshOffsets.Any())
+                {
+                    _log.Debug($"Refreshing comitted offsets: {refreshOffsets.JoinToString(", ")}");
+                    Commit(refreshOffsets, msg => Context.System.DeadLetters.Tell(msg));
+                }
+               
+                Poll();
+               
+                if (poll.Periodic)
+                    SchedulePoolTask();
+                else
+                    _delayedPoolInFlight = false;
+            }
+            else
+            {
+                // Message was enqueued before a restart - can be ignored
+                _log.Debug("Ignoring Poll message with stale target ref");
+            }
+        }
+
+        private void Poll()
+        {
+            var currentAssignment = _consumer.Assignment;
+            var initialRebalanceInProcess = _rebalanceInProgress;
+
+            try
+            {
+                if (_requests.IsEmpty())
+                {
+                    // no outstanding requests so we don't expect any messages back, but we should anyway
+                    // drive the KafkaConsumer by polling to handle partition events etc.
+                    // TODO: Make consumer pause-resume calls work
+                    // _consumer.Pause(currentAssignment);
+                    var message = _consumer.Consume(TimeSpan.FromMilliseconds(1));
+                    if (message != null)
+                        throw new IllegalActorStateException("Got unexpected Kafka message");
+                }
+                else
+                {
+                    // resume partitions to fetch
+                    IImmutableSet<TopicPartition> partitionsToFetch = _requests.Values.SelectMany(v => v.Topics).ToImmutableHashSet();
+                    /*
+                    // TODO: Make consumer pause-resume calls work
```

Comment:
@Aaronontheweb This is a most tricky thing left in this actor.
For some reason, if I will uncomment `_consumer.Resume()` call the following `_consumer.Consume()` will reset consumer's offset to 0, returning the first message in the partition. If we do not use pause-resume stuff, all works fine - at least until we will need to switch between different partitions, which is not the case so far but will be required when actor is shared between different stages.
I have found an [issue](https://github.com/confluentinc/confluent-kafka-dotnet/issues/672) with confluent driver, but it is marked as resolved for almost a year. I spent some time on it, and decided to get back as a separate issue/PR.

So far, I have commented out all pause-resume lines from implementation, so instead of initially pausing assigned partitions and resuming them on-demand, consumer actor is now consuming all assigned partitions all the time. If used will changing partitions set, this can lead to skipping messages from partitions, that do not currently have request for (such partitions will be just consumed and results will be ignored).

Reply:
All right, after fixing CI tests failing I have realized that we do not require pause-resume offset that much: it is possible to reset consumer's position each time we consume message from wrong partition. I updated implementation, so now it basically should work even with different partition sets. Sure, not consuming would be better then consuming and putting offset back, but... That is what I have right now.

Reply:
@IgorFedchenko Can you create an issue for the Confluent driver if you think it's a bug with their code? Would love to see that cross-referenced somewhere on this repository too.